catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/minio-component.adoc

```diff
@@ -0,0 +1,162 @@
+[[minio-component]]
+= Minio Storage Service Component
+:docTitle: Minio Storage Service
+:artifactId: camel-minio
+:description: Store and retrie objects from Minio Storage Service using Minio SDK.
+:since: 3.5
+:supportLevel: Preview
+:component-header: Both producer and consumer are supported
+
+*Since Camel {since}*
+
+*{component-header}*
+
+
+// component options: START
+The Minio Storage Service component supports 45 options, which are listed below.
+
+
+
+[width="100%",cols="2,5,^1,2",options="header"]
+|===
+| Name | Description | Default | Type
+| *autoCreateBucket* (common) | Setting the autocreation of the bucket if bucket name not exist. | true | boolean
+| *bypassGovernanceMode* (common) | Set this flag if you want to bypassGovernanceMode when deleting a particular object. | false | boolean
+| *configuration* (common) | The component configuration |  | MinioConfiguration
+| *customHttpClient* (common) | Set custom HTTP client for authenticated access. |  | OkHttpClient
+| *endpoint* (common) | Endpoint can be an URL, domain name, IPv4 address or IPv6 address. |  | String
+| *minioClient* (common) | Reference to a Minio Client object in the registry. |  | MinioClient
+| *objectLock* (common) | Set when creating new bucket. | false | boolean
+| *pojoRequest* (common) | If we want to use a POJO request as body or not. | false | boolean
```

Comment:
@Nayananga if two types of the requests are only possible, it's ok, but if you would like to extend, the requestType enum is better to use. It also common recommendation to review a large number of boolean flags which can be grouped together to the some enums types. Usually the minio API and Camel component pararameters can be not the same types and names.

Reply:
Hi @dmvolod, are you suggesting that I can group boolean flags with the same default value together to some enums types?
and also 

> Usually the minio API and Camel component pararameters can be not the same types and names.

could you provide me with an example as such?